src/hugr/views.rs

```diff
@@ -303,6 +305,29 @@ pub trait HugrView: sealed::HugrInternals {
     }
 }
 
+/// A view of the whole Hugr.
+/// (Just provides static checking of the type of the root node)
+pub struct WholeHugrView<'a, Root = Node>(&'a Hugr, PhantomData<Root>);
+
+impl<'a, Root: NodeHandle> WholeHugrView<'a, Root> {
+    /// Create a hierarchical view of a whole HUGR
+    ///
+    /// # Errors
+    /// Returns [`HugrError::InvalidNode`] if the root isn't a node of the required [OpTag]
+    pub fn try_new(hugr: &'a Hugr) -> Result<Self, HugrError> {
```

Comment:
impl `TryInto<WholeHugrView<'a, R>> for &'a Hugr` ?

Reply:
Nice!! :-)